ovos_PHAL_plugin_system/__init__.py

```diff
@@ -25,11 +34,107 @@ def __init__(self, bus=None, config=None):
+        if os.path.isfile(IdentityManager.OLD_IDENTITY_FILE):
+            os.remove(IdentityManager.OLD_IDENTITY_FILE)
+        if os.path.isfile(IdentityManager.IDENTITY_FILE):
+            os.remove(IdentityManager.IDENTITY_FILE)
+
+        wipe_cache = message.data.get("wipe_cache", True)
+        if wipe_cache:
+            p = f"{xdg_cache_home()}/{get_xdg_base()}"
+            if os.path.isdir(p):
+                shutil.rmtree(p, ignore_errors=True)
+
+        wipe_data = message.data.get("wipe_data", True)
+        if wipe_data:
+            p = f"{xdg_data_home()}/{get_xdg_base()}"
+            if os.path.isdir(p):
+                shutil.rmtree(p, ignore_errors=True)
+
+            # misc json databases from offline/personal backend
+            for j in ["ovos_device_info",
+                      "ovos_oauth",
+                      "ovos_oauth_apps",
+                      "ovos_devices",
+                      "ovos_metrics",
+                      "ovos_preferences",
+                      "ovos_skills_meta"]:
+                p = JsonStorageXDG(j).path
+                if os.path.isfile(p):
+                    os.remove(p)
+            for j in ["ovos_metrics",
+                      "ovos_utterances",
+                      "ovos_wakewords"]:
+                p = JsonDatabaseXDG(j).db.path
+                if os.path.isfile(p):
+                    os.remove(p)
+
+        wipe_logs = message.data.get("wipe_logs", True)
+        if wipe_logs:
+            p = f"{xdg_state_home()}/{get_xdg_base()}"
+            if os.path.isdir(p):
+                shutil.rmtree(p, ignore_errors=True)
+
+        wipe_cfg = message.data.get("wipe_configs", True)
+        if wipe_cfg:
+            if os.path.isfile(OLD_USER_CONFIG):
+                os.remove(OLD_USER_CONFIG)
+            if os.path.isfile(USER_CONFIG):
+                os.remove(USER_CONFIG)
+            if os.path.isfile(WEB_CONFIG_CACHE):
+                os.remove(WEB_CONFIG_CACHE)
+
+        reset_phal = message.data.get("reset_hardware", True)
+        if reset_phal and len(self.factory_reset_plugs):
+            reset_plugs = []
+            event = Event()
+
+            def on_done(message):
+                nonlocal reset_plugs, event
+                sid = message.data["skill_id"]
+                if sid not in reset_plugs:
+                    reset_plugs.append(sid)
+                if all([s in reset_plugs for s in self.factory_reset_plugs]):
+                    event.set()
+
+            self.bus.on("system.factory.reset.phal.complete", on_done)
+            self.bus.emit(message.forward("system.factory.reset.phal"))
+            event.wait(timeout=60)
+            self.bus.remove("system.factory.reset.phal.complete", on_done)
+
+        script = message.data.get("script", True)
+        if script:
+            script = os.path.expanduser(self.config.get("reset_script", ""))
+            if os.path.isfile(script):
+                subprocess.call(script, shell=True)
```

Comment:
@j1nx this is where any platform specific code would be executed

Reply:
so under PHAL config under this plugin section just need to add `"reset_script": "path/to/custom/code.sh"`

Reply:
And then it only just rans what is in there?

Reply:
Yes

Reply:
Not only, the bus message can request specific actions only